src/AspNetCoreModuleV2/CommonLib/Environment.cpp

```diff
@@ -0,0 +1,29 @@
+#include "Environment.h"
+
+#include <Windows.h>
+
+std::wstring
+Environment::ExpandEnvironmentVariables(const std::wstring & str)
+{
+    DWORD requestedSize = ExpandEnvironmentStringsW(str.c_str(), nullptr, 0);
+    if (requestedSize == 0)
+    {
+        throw std::system_error(GetLastError(), std::system_category(), "ExpandEnvironmentVariables");
+    }
+
+    std::wstring expandedStr;
```

Comment:
I've seen us use std::unique_ptr<wchar[]> and std::wstring interchangeably for calling windows functions. Let's pick one and stick with it?

Reply:
wchar[] is used only when it gets turned into STRU afterwards, otherwise wstring should be used.